Emby.Server.Implementations/Cryptography/CryptographyProvider.cs

```diff
@@ -107,35 +107,42 @@ public byte[] ComputeHash(string HashMethod, byte[] bytes, byte[] salt)
             }
             else
             {
-                throw new CryptographicException(String.Format("Requested hash method is not supported: {0}", HashMethod));
+                throw new CryptographicException($"Requested hash method is not supported: {HashMethod}"));
             }
-        }
+        }
 
         public byte[] ComputeHashWithDefaultMethod(byte[] bytes, byte[] salt)
         {
             return PBKDF2(DefaultHashMethod, bytes, salt, defaultiterations);
         }
         
         public byte[] ComputeHash(PasswordHash hash)
-        {
-            int iterations = defaultiterations;
-            if (!hash.Parameters.ContainsKey("iterations"))
-            {
-                hash.Parameters.Add("iterations", defaultiterations.ToString());
-            }
-            else
-            {
-                try { iterations = int.Parse(hash.Parameters["iterations"]); }
-                catch (Exception e) { iterations = defaultiterations; throw new Exception($"Couldn't successfully parse iterations value from string:{hash.Parameters["iterations"]}", e); }
+        {
+            int iterations = defaultiterations;
+            if (!hash.Parameters.ContainsKey("iterations"))
+            {
+                hash.Parameters.Add("iterations", defaultiterations.ToString(CultureInfo.InvariantCulture));
+            }
+            else
+            {
+                try
+                {
+                    iterations = int.Parse(hash.Parameters["iterations"]);
+                }
+                catch (Exception e)
+                {
+                    iterations = defaultiterations;
+                    throw new Exception($"Couldn't successfully parse iterations value from string:{hash.Parameters["iterations"]}", e);

```

Comment:
Can this be of a more descriptive exception class?

Reply:
Sure. InvalidDataException seemed appropriate so used that. Fixed.